libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigator.kt

```diff
@@ -86,7 +87,7 @@ interface MapboxNativeNavigator {
     suspend fun setRoute(
         route: DirectionsRoute?,
         legIndex: Int = INDEX_FIRST_LEG
-    ): Boolean
+    ): RouteInitInfo?
```

Comment:
Can we keep this as returning Boolean?

Unclear why RouteInitInfo should be here, because we should be able to get all the contents outside of a main function like setRoute.

Reply:
We're distributing the list of `RouteAlert`s from `RouteInitInfo` (if present) in the trip session based on this return type.

Reply:
Yeah I can see that, but why are we distributing RouteAlert's from RouteInitInfo? Seem like we could distribute voice commands from there too (not advocating that we should be, I'm just not seeing why RouteAlerts need to.

Also not defending the Boolean return, we're [not using it anyways](https://github.com/mapbox/mapbox-navigation-android/blob/ba835489d9d8c22c9397ec2b02fa07d1fd73367c/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L77).

Reply:
We are distributing `RouteAlert`s from `RouteInitInfo` to let the users query (or listen to in this case) all of the route alerts that are along the route, without having to start the trip session or wait for the first `RouteProgress`. They are available as soon as the route is parsed for the route overview use-cases.

We were discussing passing voice and banner instructions in here as well but opted not to for now https://github.com/mapbox/mapbox-navigation-native/pull/1640#discussion_r482836738.

Reply:
Was considering distributing the `RouteAlert` with another approach like listeners, but the listener are better off in MapboxTripSession. 👍  to this approach.

After thinking about it, may make sense to put the `route` and `routeBufferGeoJson` into the `RouteInitInfo`. That way you we will also clear the RouteAlerts 
https://github.com/mapbox/mapbox-navigation-android/blob/5d022f09807393210b809a579d6e6700315d98d4/libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt#L64-L65

Reply:
Yea, we could look into storing them as one in the future :+1: For now, the `RouteInitInfo` just tries to wrap the `RouteInfo` native object.

Reply:
👍